framework/core/swapchain.h

```diff
@@ -79,7 +79,7 @@ class Swapchain
 	          const uint32_t                        image_count       = 3,
 	          const VkSurfaceTransformFlagBitsKHR   transform         = VK_SURFACE_TRANSFORM_IDENTITY_BIT_KHR,
 	          const VkPresentModeKHR                present_mode      = VK_PRESENT_MODE_FIFO_KHR,
-	          const std::set<VkImageUsageFlagBits> &image_usage_flags = {VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT, VK_IMAGE_USAGE_TRANSFER_SRC_BIT});
```

Comment:
We use `VK_IMAGE_USAGE_TRANSFER_SRC_BIT` for allowing us to copy the contents of the swapchain images to disk storage. Removing this flag breaks our system tests, could I propose we have all three flags by default?